static/app/components/hovercard.tsx

```diff
@@ -65,82 +73,44 @@ type Props = DefaultProps & {
    * Popper Modifiers
    */
   modifiers?: PopperProps['modifiers'];
-};
-
-type State = {
-  visible: boolean;
-};
-
-class Hovercard extends React.Component<Props, State> {
-  static defaultProps: DefaultProps = {
-    displayTimeout: 100,
-    position: 'top',
-  };
-
-  constructor(args: Props) {
-    super(args);
-
-    let portal = document.getElementById('hovercard-portal');
-    if (!portal) {
-      portal = document.createElement('div');
-      portal.setAttribute('id', 'hovercard-portal');
-      document.body.appendChild(portal);
-    }
-    this.portalEl = portal;
-    this.tooltipId = domId('hovercard-');
-    this.scheduleUpdate = null;
-  }
-
-  state: State = {
-    visible: false,
-  };
-
-  componentDidUpdate(prevProps: Props) {
-    const {body, header} = this.props;
-
-    if (body !== prevProps.body || header !== prevProps.header) {
-      // We had a problem with popper not recalculating position when body/header changed while hovercard still opened.
-      // This can happen for example when showing a loading spinner in a hovercard and then changing it to the actual content once fetch finishes.
-      this.scheduleUpdate?.();
-    }
-  }
+  children: React.ReactNode;
+}
 
-  portalEl: HTMLElement;
-  tooltipId: string;
-  hoverWait: number | null = null;
-  scheduleUpdate: (() => void) | null;
+function Hovercard(props: HovercardProps): React.ReactElement {
+  const [visible, setVisible] = React.useState(false);
 
-  handleToggleOn = () => this.toggleHovercard(true);
-  handleToggleOff = () => this.toggleHovercard(false);
+  const inTimeout = React.useRef<number | null>(null);
+  const scheduleUpdateRef = React.useRef<(() => void) | null>(null);
 
-  toggleHovercard = (visible: boolean) => {
-    const {displayTimeout} = this.props;
+  const portalEl = React.useMemo(() => findOrCreatePortal(), []);
+  const tooltipId = React.useMemo(() => domId('hovercard-'), []);
 
-    if (this.hoverWait) {
-      clearTimeout(this.hoverWait);
+  React.useEffect(() => {
+    // We had a problem with popper not recalculating position when body/header changed while hovercard still opened.
+    // This can happen for example when showing a loading spinner in a hovercard and then changing it to the actual content once fetch finishes.
+    if (scheduleUpdateRef.current) {
+      scheduleUpdateRef.current();
     }
-
-    this.hoverWait = window.setTimeout(() => this.setState({visible}), displayTimeout);
-  };
-
-  render() {
-    const {
-      bodyClassName,
-      containerClassName,
-      className,
-      header,
-      body,
-      position,
-      show,
-      tipColor,
-      tipBorderColor,
-      offset,
-      modifiers,
-    } = this.props;
-
-    // Maintain the hovercard class name for BC with less styles
-    const cx = classNames('hovercard', className);
-    const popperModifiers: PopperProps['modifiers'] = {
+  }, [props.body, props.header]);
```

Comment:
should we destructure the props? 🤷‍♀️

Reply:
I dont think so - this way it's easy to tell that body and header are both passed down from the props. If we had [body, header] it would imply that those could have been defined locally

Reply:
that's a good argument :)

Reply:
Thank you :)